fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobRunner.java

```diff
@@ -556,9 +558,26 @@ private void doCommit(boolean isReplay) {
                 throw new DdlException("partition type " + partitionInfo.getType() + " is not supported");
             }
 
-            long sumTargetRows = job.getTmpPartitionIds().stream()
-                    .mapToLong(p -> targetTable.mayGetPartition(p).stream().mapToLong(Partition::getRowCount).sum())
-                    .sum();
+            long sumTargetRows = 0;
+            if (insertStmt != null) {
+                TransactionState txnState = GlobalStateMgr.getCurrentState()
+                        .getGlobalTransactionMgr()
+                        .getTransactionState(dbId, insertStmt.getTxnId());
+                if (txnState != null && txnState.getTxnCommitAttachment() != null) {
+                    TxnCommitAttachment attachment = txnState.getTxnCommitAttachment();
+                    if (attachment instanceof InsertTxnCommitAttachment) {
+                        sumTargetRows = ((InsertTxnCommitAttachment) attachment).getLoadedRows();
+                    }
+                }
+            }
+            
+            if (sumTargetRows == 0) {
+                LOG.warn("TxnCommitAttachment is null or invalid, fallback to partition.getRowCount()");
```

Comment:
it would great if you can attach the context including table_id/partition_id/tablet_id to the log